packages/payment-backend/src/service.ts

```diff
@@ -299,14 +315,15 @@ export const getPaymentUrl = async (p: {
   namespace: string
   orderId: string
   user: string
+  gateway: string
 }): Promise<string> => {
-  const { namespace, orderId, user: userId } = p
+  const { namespace, orderId, user: userId, gateway } = p
   if (!process.env.PAYMENT_BACKEND_URL) {
     throw new Error('No payment API backend URL set')
   }
 
-  const url = `${process.env.PAYMENT_BACKEND_URL}/payment/online/url`
-
+  let paymentMethodPart = createPaymentMethodPartFromGateway(gateway)
```

Comment:
It would be good to always use const when we have variables that are not supposed to change in the flow. Just a small "best practices" thing.